exec-unittests

```diff
@@ -102,25 +132,45 @@ _subtst_list="tool-scripts/datalog tool-scripts/postprocess tool-scripts util-sc
 let count=0
 for _major in ${major_list}; do
     if [[ "${_major}" == "agent" ]]; then
+        # The parallel program is really cool, but what a pain that it posts this
+        # citation request.
+        if [[ -z "${HOME}" ]]; then
+            # If `HOME` is not set, the `parallel` program looks here. Don't ask.
+            _para_dir=/tmp/.parallel
+        else
+            _para_dir=${HOME}/.parallel
+        fi
+        mkdir -p ${_para_dir}
+        touch ${_para_dir}/will-cite
+
+        # Count what we will do since an implicit sub-shell is used when
+        # piping `parallel` below.
         for _subtst in ${_subtst_list}; do
             if [[ -z "${subtst}" || "${subtst}" == "$(basename ${_subtst})" ]]; then
                 (( count++ ))
-                run_legacy ${_major} ${_subtst} ${posargs} || rc=1
             fi
         done
+        for _subtst in ${_subtst_list}; do
+            if [[ -z "${subtst}" || "${subtst}" == "$(basename ${_subtst})" ]]; then
+                echo ${_major} ${_subtst} ${posargs}
+            fi
+        done | parallel -k ${para_jobs_arg} para_run_legacy
```

Comment:
You can't, which is what the comment refers to.  If you combine the loops the count is lost because it is executed in a sub-shell.

Reply:
Well, gosh!

OK, I propose more specific wording for the comment, then.  E.g., something like
```
        # Count what we _will_ do since the `for` command which drives the
        # `parallel` command below will be run in a sub-shell (because it is
        # part of a "pipeline") which will make changes to `count` unavailable.
```

Alternatively, we could build the list and count it in a single loop, and then invoke `parallel` separately:
```
        let count=0
        test_list=""
        for _subtst in ${_subtst_list}; do
            test_list+="${_major} ${_subtst} ${posargs}"$'\n'
            (( count++ ))
        done
        parallel --lb -k ${para_jobs_arg} para_run_legacy <<< ${test_list}
```